### ISSUE TITLE
doesn't work with tab groups

### ISSUE THREAD

**Issue by syliew** [traxium/tabtree#16]
i didn't see any issue regarding this. do you intend to make it work or no?

**Comment by traxium** [comment]
There is a bug (https://github.com/traxium/tabtree/issues/12 and https://github.com/traxium/tabtree/issues/13) that in order to fix it requires rewriting this add-on almost from the beginning. And I've spent several hours reading Firefox source code trying to understand how Tab Groups work. And I understood that a huge work has to be done to implement Tab Groups support in this add-on. Therefore it's not rational to implement Tab Groups support in this add-on and after that one more time in the new add-on. All major missing features that require a lot of time to implement (collapsible subtrees, enhanced tab search and Tab Group support) are going to be added in this new add-on. That's my plans for now.
By the way, It's said in the add-on description on https://addons.mozilla.org/en-US/firefox/addon/tab-tree/ - "- You can't use "Tab Groups (Panorama)" with this extension"

**Comment by Ariloum** [comment]
It is sad it doesnt planning to suppot tab groups, I just tried tab tree it and it looks great. What a pity that I have to remove it only in that case. Is it really that hard to build a tree only based on tabs that currently active on selected group instead of building it on the whole tabs list?

**Comment by Ariloum** [comment]
Btw I didnt know that firefox added tab groups support, need to check it out. Im using this plugin for groups https://addons.mozilla.org/en-us/firefox/addon/tab-groups-helper/?src=ss

**Comment by traxium** [comment]
Tab Groups Helper looks much better to me then the native tab groups.
The story about tab groups support is that I completely forgot about tab groups existence while was developing Tab Tree (though I used them in the past). And I remembered about it after I published the first version on AMO when the first reviewer mentioned it. It would be much easy to integrate tab groups support at an early stage of development.

**Comment by Ariloum** [comment]
That is true, some bugs appears due to groups using, it needs to be checked why did it happens. Maybe tab indexes were changed or whatever else, I'm not a kind of ff plugins developer, I'm just a java ee developer = cant help ю much with that.

**Comment by tumpio** [comment]
Since tab groups are not supported, maybe a warning should be shown to users on first run? And check if user has tab groups.

I believe this happens when tab tree is installed while using tab groups:
https://github.com/traxium/tabtree/issues/43

**Comment by mythmon** [comment]
Tab groups are slated to be removed in a future version of Firefox in [bug 1221050](https://bugzilla.mozilla.org/show_bug.cgi?id=1221050). Given that, it might not be the best investment of time to add support for this.

**Comment by rkrug** [comment]
I think Tab Groups will stay even if removed from Firefox, there is a plugin (https://addons.mozilla.org/en-us/firefox/addon/tab-groups-panorama/) which is replacing it.

As I said in a different thread, I am using Tree-style-tab in combination with groups and it is a very nice combo.

**Comment by Luckz** [comment]
I also noticed the incompatibility while testing https://addons.mozilla.org/en-US/firefox/addon/tab-groups-panorama/ with https://addons.mozilla.org/en-US/firefox/addon/tab-groups-helper/

Unfortunately it also does not work with https://addons.mozilla.org/en-US/firefox/addon/tabgroups-manager-revived
